File: c/darts/src/darts.c
#include "darts.h"

#include <math.h>

// typedef struct {
//    float x;
//    float y;
//} coodinate_t;

uint8_t score(coordinate_t p) {
    double d = sqrt(p.x * p.x + p.y * p.y);
    if (d > 10) {
        return 0;
    } else if (d > 5) {
        return 1;
    } else if (d > 1) {
        return 5;
    } else {
        return 10;
    }
}

File: go/darts/darts.go
package darts

import (
	"math"
)

func Score(x float64, y float64) int {
	v := math.Sqrt(x*x + y*y)

	switch {
	case v <= 1.0:
		return 10
	case v > 1.0 && v <= 5.0:
		return 5
	case v > 5.0 && v <= 10.0:
		return 1
	default:
		return 0
	}
}
